static/app/components/events/autofix/autofixHighlightPopup.tsx

```diff
@@ -0,0 +1,442 @@
+  isLoading?: boolean;
+}
+
+function useCommentThread({groupId, runId}: {groupId: string; runId: string}) {
+  const api = useApi({persistInFlight: true});
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: (params: {
+      message: string;
+      retain_insight_card_index: number | null;
+      selected_text: string;
+      step_index: number;
+      thread_id: string;
+    }) => {
+      return api.requestPromise(`/issues/${groupId}/autofix/update/`, {
+        method: 'POST',
+        data: {
+          run_id: runId,
+          payload: {
+            type: 'comment_thread',
+            message: params.message,
+            thread_id: params.thread_id,
+            selected_text: params.selected_text,
+            step_index: params.step_index,
+            retain_insight_card_index: params.retain_insight_card_index,
+          },
+        },
+      });
+    },
+    onSuccess: _ => {
+      queryClient.invalidateQueries({queryKey: makeAutofixQueryKey(groupId)});
+    },
+    onError: () => {
+      addErrorMessage(t('Something went wrong when sending your comment.'));
+    },
+  });
+}
+
+function AutofixHighlightPopup({
+  selectedText,
+  groupId,
+  runId,
+  stepIndex,
+  retainInsightCardIndex,
+  referenceElement,
+}: Props) {
+  const {mutate: submitComment} = useCommentThread({groupId, runId});
+  const [comment, setComment] = useState('');
+  const popupRef = useRef<HTMLDivElement>(null);
+  const [position, setPosition] = useState({left: 0, top: 0});
+  const [threadId] = useState(() => {
+    const timestamp = Date.now();
+    const random = Math.floor(Math.random() * 10000);
+    return `thread-${timestamp}-${random}`;
+  });
+  const [optimisticMessages, setOptimisticMessages] = useState<OptimisticMessage[]>([]);
+  const messagesEndRef = useRef<HTMLDivElement>(null);
+
+  // Fetch current autofix data to get comment thread
+  const autofixData = useAutofixData({groupId});
+  const currentStep = autofixData?.steps?.[stepIndex];
+  const commentThread =
+    currentStep?.active_comment_thread?.id === threadId
+      ? currentStep.active_comment_thread
+      : null;
+  const messages = useMemo(
+    () => commentThread?.messages ?? [],
+    [commentThread?.messages]
+  );
+
+  // Combine server messages with optimistic ones
+  const allMessages = useMemo(
+    () => [...messages, ...optimisticMessages],
+    [messages, optimisticMessages]
+  );
+
+  const truncatedText =
+    selectedText.length > 70
+      ? selectedText.slice(0, 35).split(' ').slice(0, -1).join(' ') +
+        '... ...' +
+        selectedText.slice(-35)
+      : selectedText;
+
+  const currentUser = useUser();
+
+  useLayoutEffect(() => {
+    if (!referenceElement || !popupRef.current) {
+      return undefined;
+    }
+
+    const updatePosition = () => {
+      const rect = referenceElement.getBoundingClientRect();
+      setPosition({
+        left: rect.left - 320,
+        top: rect.top,
+      });
```

Comment:
might consider wrapping this with startTransition https://react.dev/reference/react/startTransition